### PR TITLE
Add test cases for operations between complex and real numbers

### PR DESCRIPTION
**Pull request by dawidl022**
As discussed here: https://github.com/exercism/problem-specifications/issues/1852

Added one test for each of the 4 basic operations: addition, subtraction, multiplication and division. The first operand is a complex number and the second is an integer.

### CODE REVIEW

File: exercises/complex-numbers/canonical-data.json [hunk 1]
```diff
@@ -363,6 +363,51 @@
           "expected": [-2, 0]
         }
       ]
+    },
+    {
+      "description": "Operations between real numbers (int/float) and complex numbers",
```
Comment (SaschaMann) [inline exercises/complex-numbers/canonical-data.json:368]:
```suggestion
      "description": "Operations between real numbers and complex numbers",
```

Number types vary between languages, so I think it's best to stick to `real numbers` here and let the tracks map that to whatever type fits their language.

**Review by SaschaMann — COMMENTED** [review]
Thanks for adding these!

I don't know if it makes a difference for any tracks but it might be useful to have the same tests in reverse too, so subtracting a complex number from a real number etc. Or perhaps that should be left to the implementation/track maintainers. I'm not sure.

**Comment by dawidl022** [comment]
> I don't know if it makes a difference for any tracks but it might be useful to have the same tests in reverse too, so subtracting a complex number from a real number etc. Or perhaps that should be left to the implementation/track maintainers. I'm not sure.

I am honestly not sure how to implement this the other way around (in python anyways). For example I am able to get `my_complex_number * 5` to work no problem, but not `5 * my_complex_number`.

**Comment by SaschaMann** [comment]
> I am honestly not sure how to implement this the other way around (in python anyways).

That sounds like a good reason to have them the other way around as a commutativity check.

**Comment by dawidl022** [comment]
> IIRC you could implement `__rmul__(self, x)` (see [docs](https://docs.python.org/3/reference/datamodel.html#object.__rmul__)).

Thank you! I did not know these methods existed.

> That sounds like a good reason to have them the other way around as a commutativity check.

Yes, I agree. It would help people like me to learn a new thing along the way.

**Review by wolf99 — APPROVED** [review]

**Pull request merged**